homeassistant/components/tessie/sensor.py

```diff
@@ -41,6 +41,20 @@ class TessieSensorEntityDescription(SensorEntityDescription):
 
 
 DESCRIPTIONS: tuple[TessieSensorEntityDescription, ...] = (
+    TessieSensorEntityDescription(
+        key="charge_state_charging_state",
+        icon="mdi:ev-station",
+        options=[
+            "starting",
+            "charging",
+            "stopped",
+            "complete",
+            "disconnected",
+            "nopower",
+        ],
+        device_class=SensorDeviceClass.ENUM,
+        value_fn=lambda x: x.lower() if isinstance(x, str) else x,
```

Comment:
Why is the instance check and lowering needed?

Reply:
Translation keys can only be lowercase.

You can run .lower() on a Sensor Value which can be an int.

Reply:
> Translation keys can only be lowercase.
> 
> You can't run .lower() on a Sensor Value which can be an int. (Even though it never will be anything but one of the Enum values in this integration)

When an int value is returned as value, it will raise an error as one of the options is expected.
I would suggest refactoring the code so that we can avoid the instance check and only return string values.

Reply:
Ok it looks like translation keys are case insensitive, so I can remove the value_fn all together.